linear-regression/helper_functions.py

```diff
@@ -0,0 +1,278 @@
+import pandas as pd
+from sklearn.metrics import mean_squared_error, mean_absolute_error, r2_score
+from sklearn.preprocessing import PowerTransformer
+from typing import Tuple
+import numpy as np
+import pytest
+
+
+def create_equatorial_distance_feature(df):
+    df['distance_from_equator'] = abs(df['Latitude'])
+    df['distance_from_equator_bin'] = pd.cut(df['distance_from_equator'], bins=10, labels=False)
+    return df
+
+
+def evaluate_model(y_true, y_pred):
+    mse = mean_squared_error(y_true, y_pred)
+    mae = mean_absolute_error(y_true, y_pred)
+    r2 = r2_score(y_true, y_pred)
+    return mse, mae, r2
+
+
+def linear_regression_pipeline(
+    X_train, X_test, y_train, y_test, target_transformer=None
+):
+    from sklearn.linear_model import LinearRegression
+    lr = LinearRegression()
+    lr.fit(X_train, y_train)
+
+    train_preds = lr.predict(X_train)
+    test_preds = lr.predict(X_test)
+
+    if target_transformer:
+        y_train = target_transformer.inverse_transform(y_train)
+        train_preds = target_transformer.inverse_transform(train_preds)
+        y_test = target_transformer.inverse_transform(y_test)
+        test_preds = target_transformer.inverse_transform(test_preds)
+
+
+    train_metrics = evaluate_model(y_train, train_preds)
+    test_metrics = evaluate_model(y_test, test_preds)
+
+    return lr, train_metrics, test_metrics
+
+
+def scale_features(X_train, X_test, feature_cols):
+
+    from sklearn.preprocessing import StandardScaler
+
+    # create a StandardScaler object and fit it to the training data with the specified feature columns
+    ss = StandardScaler().fit(X_train[feature_cols])
+
+    # transform the feature columns in both the training and testing datasets
+    X_train[feature_cols] = ss.transform(X_train[feature_cols])
+    X_test[feature_cols] = ss.transform(X_test[feature_cols])
+
+    # return the scaled training and testing datasets
+    return X_train[feature_cols], X_test[feature_cols]
```

Comment:
You should return X_train and X_test without subsetting in case feature_cols doesn't include all columns